src/cls/rgw/cls_rgw_client.h

```diff
@@ -104,6 +104,7 @@ class BucketIndexAioManager {
     if (r >= 0) {
       add_pending(arg->id, c, oid);
     } else {
+      arg->put();
```

Comment:
does this also apply to `aio_operate(librados::IoCtx& io_ctx, const string& oid, librados::ObjectWriteOperation *op)` ?

Reply:
the aforementioned function put the op into the internal queue, and if it is processed, the callback function is called, inside of which the callback argument is released. But if the error occurs before being put into the queue, it just returns with error code, and there is no chance to release the resources.

Reply:
@tchaikov I'm sorry I got it wrong. You are right. Same for aio_operate for writing. I'll push it again.

Reply:
thanks!